src/utils/cluster_utils.cpp

```diff
@@ -44,40 +40,44 @@ State bayesmix::cluster_estimate(BaseCollector *coll) {
   // Find iteration with the least error
   std::ptrdiff_t ibest;
   unsigned int min_err = errors.minCoeff(&ibest);
-  return chain[ibest];
+  return allocation_chain.row(ibest).transpose();
 }
 
 //! \param filename Name of file to write to
 void bayesmix::write_clustering_to_file(
-    const State &best_clust,
+    const Eigen::VectorXi &best_clust,
     const std::string &filename /*= "resources/clust_best.csv"*/) {
+
+  // TODO and Discuss !
```

Comment:
Writing a matrix of integers to a file is as easy as it sounds, however the user may want to see the actual unique values for the clusters. Should we implement this?

Reply:
I don't think one wants to see the unique values at the best clustering iteration (it would be a non-bayesian estimate), but rather estimate them from the full posterior once you fix the best clustering.